src/integrator/integrator.h

```diff
@@ -63,20 +66,21 @@ void Integrator::PredictorCorrector<soltype>::solve(
     const log_level_t log_level) const
 {
   for(int step = 0; step < time_idx_ubound; ++step) {
-    solve_step(step);
+    solve_step(step, num_corrector_steps);
     if(log_level >= log_level_t::LOG_INFO) log_percentage_complete(step);
   }
 }
 
 template <class soltype>
-void Integrator::PredictorCorrector<soltype>::solve_step(const int step) const
+void Integrator::PredictorCorrector<soltype>::solve_step(
+    const int step, const int num_corrector_steps) const
```

Comment:
You've already added `num_corrector_steps` as a class variable which gives `solve_step` access to it automatically. You don't need to pass it as an explicit argument.

```suggestion
     solve_step(step); 
     if(log_level >= log_level_t::LOG_INFO) log_percentage_complete(step); 
   } 
 } 

 template <class soltype> 
 void Integrator::PredictorCorrector<soltype>::solve_step(const int step) const
```